### ISSUE TITLE
Add light level to token object

### ISSUE THREAD

**Issue by txm3278** [foundryvtt/foundryvtt#10023]
### User Experience

There are many features in both D&D5e and PF2 that require a token to be in a certain light level or work differently depending on the light level (etc. Perception checks relying on sight in D&D5e). Adding the current light level of the token would allow modules/systems to make use of this without having to run expensive calculations to determine it.

**Comment by aaclayton** [comment]
We've discussed a desire to add this functionality within the team, so we are aligned with the feature request. I'm unsure on timing though, it may not be possible in V12 - but we could consider it.

**Comment by LukasPrism** [comment]
Adding to this, if there were other properties lights could impart that could be accessed also that would be really useful. For example if lights could be assigned the “sunlight” property. There are many creatures affected by sunlight, there’s even a neat weapon in A House Divided (ahem) that changes based on whether it’s in sunlight or not.

**Comment by aaclayton** [comment]
Nice feature, but deprioritizing from V12 since it falls below the line.

**Comment by Sayshal** [comment]
Would love to see this!

**Comment by mkahvi** [comment]
Is this not implemented with `canvas.effects.getDarknessLevel()` ?

**Comment by txm3278** [comment]
> Is this not implemented with `canvas.effects.getDarknessLevel()` which was added with v12?

No, this gets the scene darkness, not the light level at a token's position
